webapp/src/marked.tsx

```diff
@@ -42,7 +42,47 @@ export class MarkedContent extends data.Component<MarkedContentProps, MarkedCont
     private renderSnippets(content: HTMLElement) {
         const { parent } = this.props;
 
-        pxt.Util.toArray(content.querySelectorAll(`.lang-blocks`))
+        pxt.Util.toArray(content.querySelectorAll(`code.lang-typescript`))
+            .forEach((langBlock: HTMLElement) => {
+                const code = langBlock.textContent;
+                const wrapperDiv = document.createElement('div');
+                pxsim.U.clear(langBlock);
+                langBlock.appendChild(wrapperDiv);
+                wrapperDiv.className = 'ui segment raised';
+                const preDiv = document.createElement('pre') as HTMLPreElement;
+                preDiv.textContent = code;
+                pxt.tutorial.highlight(preDiv);
+                wrapperDiv.appendChild(preDiv);
+            });
+
+        pxt.Util.toArray(content.querySelectorAll(`code.lang-spy`))
+            .forEach((langBlock: HTMLElement) => {
+                const code = langBlock.textContent;
+                const wrapperDiv = document.createElement('div');
+                pxsim.U.clear(langBlock);
+                langBlock.appendChild(wrapperDiv);
+                wrapperDiv.className = 'ui segment raised loading';
+                if (MarkedContent.blockSnippetCache[code]) {
```

Comment:
At a glance looks this would be the reason I have to restart the local serve - will this grab the previously cached e.g. typescript output and show that instead of the intended python when changing snippet language?

Either way, unlikely to be something that ever matters outside of local serve, and trivially worked around by making a quick change to the content of the snippet that causes the code to need to be rerendered anyways

Reply:
This cache should have been cleared when closing the turorial

Reply:
Ah, you're right. I must have had a pending change that forced it to refresh when I tested it with this cache section removed.